### ISSUE TITLE
Checks Side Panel (First Pass)

### ISSUE THREAD

**Issue by katherinejensen00** [paranext/paranext-core#1283]
**User Story**
As a user, I want to be able to view the results of checks so that I can easily see all of the check failures I need to address so that I can have the most accurate/error free translation possible.

**Description**
Create a side panel for checks that can display all of the different check states. Note this is the first iteration on the checks side panel.

[Figma Link](https://www.figma.com/design/M2zusN2e1VEqH5kuD9aguD/Checks?node-id=978-5833&node-type=canvas&t=KlA5FcHAtqa38nMn-0)

**Implementation idea**

- [x] Create a side panel for checks (separate tab from editor but can be arranged next to editor)
- [x] Create Check card (note will be shared between marginal component and island component/webview which means making badge optional)
- [x] Create Check States that can actually be seen without clean up (Active, Denied) and use those check states when creating check cards to display in the side panel list (see Check States part of Figma design)
- [x] Create a menu item from the editor menu that opens the checks side panel island component/webview

Note: Fixed state will be implemented as part of the Clean up task that will create more check result list state and Checking will be implemented once a bug is fixed that currently would make it flaky.

For developing purposes on this case: hard code Repeated words as the check whose results display

![Image](https://github.com/user-attachments/assets/e374c55c-746b-4608-be26-2ad4eabe3e02)

**Comment by katherinejensen00** [comment]
@paranext/paranext-ux Should clicking on a check card take you to the spot in the Scripture text where the text that needs to be edited occurs?

**Comment by merchako** [comment]
> @paranext/paranext-ux Should clicking on a check card take you to the spot in the Scripture text where the text that needs to be edited occurs?

Yes, correct @katherinejensen00. When a card it focused, the related editor should scroll to and highlight the relevant range.

**Comment by ianhewubs** [comment]
> Yes, correct @katherinejensen00. When a card it focused, the related editor should scroll to and highlight the relevant range.

Where "relevant range" means either: 
1. Highlight the character(s) or marker in the source project/resource which was identified by the check result. 
2. If no character or marker is identified, highlight the verse(s) containing the result.

**Comment by Sebastian-ubs** [comment]
Hi, what's the scope of this issue?
Is it
- "Just a component that's not yet used in Paratext"
- "A component that's part of the shared editor webview and can interact with it?"
(or something else)?

Asking because it differs from the original [task breakdown](https://github.com/paranext/paranext-core/issues/1222#issuecomment-2426949191).
- #1233 is not added to the current sprint

Recognizing that there is no ux design showing a standalone checks tab that can be freely moved around (like the current result list). Is #1234 still needed @ianhewubs ?

**Comment by katherinejensen00** [comment]
@Sebastian-ubs This task is what we are calling the "island" component that is next to the editor. We need a separate implementation for the margin inside of the editor (see #1287). This component will be used in Paratext and other tasks will add on functionality. We are just trying to break this up into bite size pieces so more people can work on it and it is easier to get through all of the stages. 

After the UX demo on Monday, I created tasks based on the designs and didn't realize until we get into planning yesterday that there were some cases already created that were similar but maybe hadn't been updated since the demo. My goal today is to clean that up today so everything is synchronized and nothing gets lost.

**Comment by Sebastian-ubs** [comment]
Thanks @katherinejensen00, I understand how we got here. Still trying to get clarity on the outcome / scope of this story
Does
> This component **will be used** in Paratext

mean "as part of this story" or "in another story"?

So is the outcome of this story
- a component that can show check results
- that is showing the real check results (although not able to change the scope right now)
- that shows up as part of the shared editor webview (shows up always until #1309 is implemented)
- can select relevant scripture text or verse in the editor (like described in [Ian's comment](#issuecomment-2473152560))

or parts of it / something else?

**Comment by ianhewubs** [comment]
Ideally, the "island" implementation needs to be a webview and its location should be dependent on program settings. 

We expect there to be PT9 devotees who want exactly one "results list" to appear on the right side of Paratext, always (but be movable as required). However, we intend to make it possible for a results island (one per project) to appear to the side of any project (right side for LTR Paratext UI language, left side for RTL UI).

Program setting something like:
> **Results list behaviour**
> - [x]  Display check results to the side of any project window
> - [ ]  Single Results List for all projects (Paratext 9 behaviour)

Styling should be consistent. In both cases the UI is a tabbed webview which can be moved as required by the user.

**Comment by katherinejensen00** [comment]
@Sebastian-ubs This component being used means that in this story this component/webview will be able to be opened (I am assuming from the menu but I can make that more explicit in the case) and viewed and it will display the checks results list in a limited way until 1284 and 1285 are completed. Navigation is being handled in 454.

The island is a separate webview. I am asking Matt and TJ right now if it is possible to automatically open that and keep it next to the editor, but I don't think we have done anything else like that yet.

**Comment by ianhewubs** [comment]
> The island is a separate webview. I am asking Matt and TJ right now if it is possible to automatically open that and keep it next to the editor, but I don't think we have done anything else like that yet. @katherinejensen00 @tjcouch-sil @lyonsil 

If it is tricky to make the checks "island" appear next to the editor, placing it at the absolute right side of the application, full height, like in PT9 is fine. Someday we'll want to be able to place tabs next to or below other specific tabs, but if we can't do it today, that's OK. 

I expect that for many layouts, placing the checks immediately next to the editor will create an disappointingly small checks window and cause the rest of the user's layout to be squeezed in potentially annoying ways. Showing checks/results on the side of the application avoids that. (i.e. We put the Results List on the right side in Paratext 9 for several good reasons.)

**Comment by Sebastian-ubs** [comment]
according to Katherine this includes filtering by scope (visible text, chapter, book, all), but rest of filtering will follow later in https://github.com/paranext/paranext-core/issues/1285

**Comment by katherinejensen00** [comment]
Quick clarification, filtering by visible text is complicated and will be done in #1298

**Comment by jolierabideau** [comment]
@roopa0222 Testing criteria for this issue:

I just opened the PR, so it is not merged yet. The web view is currently forced to only show check results for the Repeated Words check, so if you open a project at a scripture reference with no Repeated Words check failures you will see "No check results."

- Open a project with Repeated Words check failures
    - Select the "Open checks side panel" menu item
    - See a list of check result cards
- If you click on a card it should turn white, showing that it is selected
    - The scripture reference should change to the scripture reference on that card
    - The editor should scroll to that scripture reference (sometimes this takes multiple clicks)
- Open the dropdown menu on a selected card with the ..., select "Deny"
    - The card should change to grey, and you should see a Badge saying "Denied" show up
- Open the dropdown menu again, select "Approve"
    - The card should change back to look like the others, and the "Denied" Badge should go away
- If you change the scripture reference on either the checks side panel or the editor, the checks and editor should update to match that scripture reference
